### PR TITLE
MM-T399 Invalidate Pending Email Invitations

### PR DESCRIPTION
**Pull request by josephk96**
I created a test to check that invites a user to a team, ensures the user can sign up via the email link provided, invite two more users, invalidate their invites, retrieve the email of the second user and go to the provided sign up link and validate that the correct error message is shown.

Test case: https://automation-test-cases.vercel.app/test/MM-T399

### CODE REVIEW

File: e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+
+        // # Wait half a second to ensure that the modal has been fully loaded and check if the instance is licensed
+        cy.wait(TIMEOUTS.HALF_SEC);
+        checkIfLicensed();
+
+        // # Invite the first user (the one that won't be invalidated and logout)
+        inviteNewUser(emailOne);
+        cy.wait(TIMEOUTS.FIVE_SEC);
```
Comment (saturninoabril) [inline e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:57]: I think it should click "Done" to complete the process.
```suggestion
        cy.findByText('Done').should('be.visible').click();
```

File: e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+        // # Type username and password
+        cy.get('#name').should('be.visible').type(usernameOne);
+        cy.get('#password').should('be.visible').type(password);
+
+        // # Attempt to create an account by clicking on the 'Create Account' button
+        cy.get('#createAccountButton').click();
+
+        // * Check that the display name of the team the user was invited to is being correctly displayed
+        cy.get('#headerTeamName').should('contain.text', testTeam.display_name);
```
Comment (saturninoabril) [inline e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:71]: I'm getting random failing test here which can be solved by adding a timeout option.
```suggestion
        cy.get('#headerTeamName', {timeout: TIMEOUTS.HALF_MIN}).should('contain.text', testTeam.display_name);
```

File: e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            cy.get('#sidebarItem_town-square').should('exist');
+        });
+
+        // * Check that the 'Welcome to: Mattermost' message is visible
+        cy.get('#tutorialIntroOne').findByText('Welcome to:').should('be.visible');
+        cy.get('#tutorialIntroOne').findByText('Mattermost').should('be.visible');
+
+        // # Logout from the current user and login as sysadmin
+        cy.apiLogout();
+        cy.apiAdminLogin();
```
Comment (saturninoabril) [inline e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:84]: I think it needs to reload a page after login.

File: e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+        // # Open the 'Invite People' full screen modal
+        cy.get('.sidebar-header-dropdown__icon').click();
+        cy.get('#invitePeople').find('button').eq(0).click();
+
+        // # Wait half a second to ensure that the modal has been fully loaded and check if the instance is licensed
+        cy.wait(TIMEOUTS.HALF_SEC);
+        checkIfLicensed();
```
Comment (saturninoabril) [inline e2e/cypress/integration/onboarding/invalidate_pending_email_invitations_spec.js:92]:
```suggestion
        // # Open the 'Invite People' full screen modal
        cy.findByLabelText('main menu').should('be.visible').click();
        cy.findByText('Invite People').should('be.visible').click();
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @josephk96! Please see comments.

**Comment by josephk96** [comment]
@saturninoabril ,  I updated the `team_settings/helpers.js` file to be in sync with https://github.com/mattermost/mattermost-webapp/pull/6263/files and used the methods there in this test. I hope that it would be fine, since I am in the `onboarding` directory but am importing and using functions from another directory. If it isn't, please let me know and I will refactor this. I also added the signup page visibility check in the verifyEmailInviteAndVisitLink function :)

@josephbaylon, please let me know if the test is still failing for you :)

**Comment by josephk96** [comment]
I also tested MM-T389 and MM-T386 to use the `verifyEmailInviteAndVisitLink` function with the newly included signup section visibility check and they are working

**Comment by josephk96** [comment]
Thanks @josephbaylon ! Btw, I'm a big fan of your name, it's super cool💯

**Comment by josephbaylon** [comment]
@josephk96 Hahaha. Likewise 🤣

**Review by saturninoabril — APPROVED** [review]
Thanks @josephk96!

**Pull request merged**